api/server/response_options.proto

```diff
@@ -20,7 +21,15 @@ message ConcurrencyBasedLinearDelay {
 // configuration will override.
 message ResponseOptions {
   // List of additional response headers.
-  repeated envoy.api.v2.core.HeaderValueOption response_headers = 1;
+  //
+  // Envoy deprecated its v2 API, prefer to use v3_response_headers instead.
+  // Mutually exclusive with v3_response_headers.
+  repeated envoy.api.v2.core.HeaderValueOption response_headers = 1 [deprecated = true];
```

Comment:
Similar thought here as in the other PR: the actual definitions for the HeaderValueOption fields don't seem to change. I wonder if we can leverage that somehow?

Reply:
Our goal with this and the other PR is to avoid forcing our users to import and use deprecated Envoy v2 API primitives if they don't want to. Since we exposed the v2 primitives on our API, this does seem like the best way forward. If this becomes a burden, we can start thinking about hiding the Envoy API primitives. Considering that would mean more maintenance and the fact that Envoy isn't going to perform an update like this often, we are probably better off with the current solution.